src/Configuration/Dotenv/Scheme/Parser.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Configuration.Dotenv.Scheme.Parser where
+
+import Control.Applicative
+import Data.Either
+import Data.Void
+import Text.Megaparsec
+import Text.Megaparsec.Char
+
+import Configuration.Dotenv.Scheme.Helpers
+import Configuration.Dotenv.Scheme.Types
+
+areParseable
+  :: Config
+  -> [(String, String)]
+  -> Bool
+areParseable config envs =
+  let envsWithConf = mapMatchEnvWithConf config envs
+      canBeParsed (v, t)= v `isParseableAs` t
+   in all canBeParsed envsWithConf
+
+isParseableAs
```

Comment:
Now here you could return a list `[Left error]` so that you can have messages related to the variables that could not be parsed.